static-assets/components/cstudio-admin/base.js

```diff
@@ -58,6 +64,38 @@
       );
     },
 
+    initRouter(tools) {
+      const _self = this;
+      const toolsNames = tools.map((tool) => tool.name);
+
+      routie('tool/:name?', function(name) {
+        if (toolsNames.includes(name)) {
+          _self.currentRoute = name;
+          if (_self.toolsModules[name]) {
```

Comment:
provided you already checked the name is correct above (line 72's if), is this if necessary?

Reply:
@rart no, the toolsNames list has the names (ids) of the tools that are to be loaded, the toolsModules are the ones already loaded